docs/reference/measure_writing_guide.md

```diff
@@ -1694,6 +1741,134 @@ Below is an example run method that manipulates and reports values related to a
+
+```ruby
+class CreateScheduleFile < OpenStudio::Measure::ModelMeasure
+
+  [...]
+
+  def run(model, runner, user_arguments)
+
+    # Locate the resource file (this resolves to something like './measures/resources/schedulefile.csv')
+    csv_in_path = File.join(File.dirname(__FILE__), 'schedulefile.csv')
+
+    # Instantiate an External File: this will automatically copy to first path in WorkflowJSON: `runner.workflow.filePaths[0]`, typically './generated_files/'
+    externalFile_ = OpenStudio::Model::ExternalFile::getExternalFile(model, csv_in_path)
+    if (!externalFile_)
+      runner.registerError("Failed to instantiate an External File")
+      return false
+    end
+
+    column = 1
+    rowsToSkip = 1
+    scheduleFile = OpenStudio::Model::ScheduleFile.new(externalFile_.get(), column, rowsToSkip)
+    scheduleFile.setName("ExampleScheduleFile")
+
+    return true
+end
+```
+
+#### How to output any other file
+
+To output any file you may need that isn't an `ExternalFile`, you should rely on two things, in order of preferences:
+* `runner.workflow.filePaths[0]`: this will typically resolve to `./generated_files`
+* `runner.workflow.absoluteRootDir`: this will resolve to '.'
+
+`.` is the location defined as the `root` key inside the `workflow.osw`, or if not specified the location of the `workflow.osw` itself.
+
+```ruby
+class CreateScheduleFile < OpenStudio::Measure::ModelMeasure
+
+  [...]
+
+  def run(model, runner, user_arguments)
+
+    # Locate the resource file (this resolves to something like './measures/resources/schedulefile.csv')
+    csv_in_path = File.join(File.dirname(__FILE__), 'schedulefile.csv')
+
+    # Canonical way: write to the ./generated_files directory
+    out_file = File.join(runner.workflow.filePaths[0].to_s, 'myfile.csv')
+    File.open(out_file, 'w') do |f|
+      f << "Hello"
+    end
+
+    # Prefer the above, but one valid use case would be to output to the reports/ folder
+    # (this is a *ModelMeasure*, you cannot just name it './report.html'
+    # and have it copied automatically like described above in Reporting measure section)
+
+    # either /tmp/osmodel-1622719126-1/ApplyMeasureNow
+    # or ./<model_companion_dir>/
+    rootDir = runner.workflow.absoluteRootDir.to_s
+
+    html_out_path = 'report.html'
+    if (File.basename(rootDir) == 'ApplyMeasureNow')
+      html_out_path = File.absolute_path(
+        File.join(rootDir, '..', 'resources', 'reports', html_out_path))
+    else
+      html_out_path = File.absolute_path(
+        File.join(rootDir, 'reports', html_out_path))
+    end
+    outDir = File.dirname(html_out_path)
+    if !File.exists?(outDir)
+      FileUtils.mkdir_p(outDir)
+    end
+    File.open(html_out_path, 'w') do |f|
+      f << "<html><head><title>My Custom Report that works with Apply Now!</title></head><body><h1>Hello World!</h1></body></html>"
+    end
```

Comment:
So perhaps this section is overkill or an invitation for bad ideas. I don't know. But people are definitely already doing stuff like this (cf discussion on https://github.com/NREL/OpenStudio/issues/4046) except they are not doing it right for lack of documentation.